src/main.rs

```diff
@@ -1,3 +1,196 @@
+use std::io::{self, prelude::*};
+use std::os::unix::net::UnixStream;
+use std::process::{Command, Stdio};
+use std::{thread, time};
+
+#[repr(u8)]
+enum MsgType {
+    MsgQuit = 1,
+    MsgRunProcess,
+}
+
+#[repr(u8)]
+enum MsgSubType {
+    SubMsgEnd = 0,
```

Comment:
Nitpick: IMO `MsgSubType::End` is sufficient

Reply:
I wanted it to match the stuff in `.h` as closely as possible. For all discussions about protocol please use #2